### PR TITLE
[spanmetrics] Do not drop metric data points on attributes cache miss.

### PR DESCRIPTION
**Pull request by kovrus**
**Description:** <Describe what has changed.>

Components: 
- `spanmetrics` connector 
- `spanmetrics` processor 

The `metric key to attributes` cache should improve performance but should result in data loss when on cache misses. This change addresses the issue with dropped data points.

At this point, the core logic of the components is the same, so it should be fine to fix this issue for both components in the same PR. 

**Testing:** <Describe what testing was performed and which tests were added.>
Existing unit tests

### CODE REVIEW

**Comment by kovrus** [comment]
cc: @albertteoh

**Review by albertteoh — APPROVED** [review]
👍🏼

**Comment by kovrus** [comment]
cc: @Cluas

File: connector/spanmetricsconnector/connector.go [hunk 1]
```diff
@@ -112,11 +112,25 @@ func newDimensions(cfgDims []Dimension) []dimension {
...
 
-type histogramData struct {
+type histogram struct {
+	attributes pcommon.Map
+
 	count         uint64
 	sum           float64
 	bucketCounts  []uint64
-	exemplarsData []exemplarData
+	exemplarsData []exemplar
```
Comment (Cluas) [inline connector/spanmetricsconnector/connector.go:121]: Maybe we can rename `exemplarsData` -> `exemplars`

File: connector/spanmetricsconnector/connector.go [hunk 2]
```diff
@@ -513,7 +509,7 @@ func sanitizeRune(r rune) rune {
 }
 
 // setExemplars sets the histogram exemplars.
-func setExemplars(exemplarsData []exemplarData, timestamp pcommon.Timestamp, exemplars pmetric.ExemplarSlice) {
+func setExemplars(exemplarsData []exemplar, timestamp pcommon.Timestamp, exemplars pmetric.ExemplarSlice) {
```
Comment (Cluas) [inline connector/spanmetricsconnector/connector.go:512]: same `exemplarsData`-> `exemplars`

File: processor/spanmetricsprocessor/processor.go [hunk 3]
```diff
@@ -544,7 +541,7 @@ func sanitizeRune(r rune) rune {
 }
 
 // setExemplars sets the histogram exemplars.
-func setExemplars(exemplarsData []exemplarData, timestamp pcommon.Timestamp, exemplars pmetric.ExemplarSlice) {
+func setExemplars(exemplarsData []exemplar, timestamp pcommon.Timestamp, exemplars pmetric.ExemplarSlice) {
```
Comment (Cluas) [inline processor/spanmetricsprocessor/processor.go:544]: same

**Review by Cluas — APPROVED** [review]
LGTM, nice job!

**Review by djaglowski — APPROVED** [review]

**Pull request merged**